src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
+    ):
+        super(SearchIndex, self).__init__()
+        self.design_doc = ddoc
+        if search_index_name is not None:
+            self._search_index_name = search_index_name
+        if search_index is not None:
+            self['index'] = codify(search_index)
+        self['analyzer'] = codify('standard')
```

Comment:
Why not allow the developer a chance to set the analyzer as part of the constructor?  You forcing anyone who wants to set the analyzer do it in 2 steps by not providing that option in the constructor.